controllers/operatorpolicy_status.go

```diff
@@ -1060,9 +1097,12 @@ func existingInstallPlanObj(ip client.Object, phase string) policyv1.RelatedObje
 
 	switch phase {
 	case string(operatorv1alpha1.InstallPlanPhaseRequiresApproval):
-		// FUTURE: check policy.spec.statusConfig.upgradesAvailable to determine `compliant`.
-		// For now, assume it is set to 'NonCompliant'
-		relObj.Compliant = string(policyv1.NonCompliant)
+		// Check policy.spec.statusConfig.upgradesAvailable to determine `compliant`.
+		if complianceConfig != "Compliant" {
+			relObj.Compliant = string(policyv1.NonCompliant)
+		} else {
+			relObj.Compliant = string(policyv1.Compliant)
+		}
 	case string(operatorv1alpha1.InstallPlanPhaseComplete):
 		relObj.Compliant = string(policyv1.Compliant)
 	default:
```

Comment:
Does this default mean that an InstallPlan that is in the installing state is NonCompliant? I would not have expected this if `complianceConfig` is `Compliant`.